### PR TITLE
Front end auth

### PR DESCRIPTION
**Pull request by DeeTheDev**
Initial React components and configurations
- Add `react-router-dom` for routing/navigating across routes and components
- Add `react-bootstrap` CSS framework for styling
- Add `react-phone-number-input` for phone number masking/formatting
- Configured basic server API fetch call system inside `src/api`
- Add `react-i18next` basic localization example inside `Home` component (including languageDetection and multi-translation JSON API)

Notes:
- TODO's in places where we should update once API is connected correct server path
- `OneTimePassword` component has some error handling but it's undone until API is done
- `Start` component needs to send API call (uncomment API code)
- `Onboarding` component needs error handling and API call

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,5 +4,8 @@
   },
   "scripts": {
     "build": "echo 'use Makefile to build frontends'"
+  },
+  "dependencies": {
+    "i18next-browser-languagedetector": "^6.1.3"
```
Comment (rgalanakis) [inline package.json:9]: I think this is a mistake?

File: webapp/public/index.html [hunk 2]
```diff
@@ -24,7 +24,19 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
+    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css"
```
Comment (rgalanakis) [inline webapp/public/index.html:27]: We can load this dynamically; I'll show you how in the future, this is good for now (maybe make an issue as a reminder).

File: webapp/public/index.html [hunk 3]
```diff
@@ -24,7 +24,19 @@
...
+      integrity="sha512-xodZBNTC5n17Xt2atTPuE1HxjVMSvLVW9ocqUKLsCC5CXdbqCmblAshOMAS6/keqq/sMZMZ19scR4PsZChSR7A=="
+      crossorigin=""/>
+    <script src="https://unpkg.com/leaflet@1.7.1/dist/leaflet.js"
+      integrity="sha512-XQoYMqMTK8LvdxXYG3nZ448hOEQiglfqkJs1NOQV44cWnUrBc8PkAOcXy20w0vlaXaVUearIOBhiXZ5V3ynxwA=="
+      crossorigin=""></script>
     <title>React App</title>
+    <style>
+
+#mapid {
```
Comment (rgalanakis) [inline webapp/public/index.html:36]: Can you put this into the normal stylesheets, or a css module?

File: webapp/src/api/apiRequest.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    method = 'GET'
+  } = request;
+
+  const headers = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  }
+  try {
+    // TODO: temporary json file path, switch to 'path' when server API is done
+    const fetchResponse = await fetch('jsonUserList.json', {
```
Comment (rgalanakis) [inline webapp/src/api/apiRequest.js:15]: Very clever interrim solution!

File: webapp/src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import Onboarding from './Onboarding';
+import Dashboard from './Dashboard';
+
+function App() {
+  return (
+    <Router>
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/start" element={<Start />} />
+        <Route path="/oneTimePassword" element={<OneTimePassword />} />
```
Comment (rgalanakis) [inline webapp/src/components/App.js:20]: Let's use `one-time-password` for route casing.

File: webapp/src/components/App.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import Start from './Start';
+import OneTimePassword from './OneTimePassword';
+import Onboarding from './Onboarding';
+import Dashboard from './Dashboard';
+
+function App() {
+  return (
+    <Router>
+      <Routes>
+        <Route path="/" element={<Home />} />
```
Comment (rgalanakis) [inline webapp/src/components/App.js:18]: I think you want `element={Home}` in case props need to be passed.

Comment (rgalanakis) [reply webapp/src/components/App.js:18]: Same for others.

File: webapp/src/components/Home.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import Col from 'react-bootstrap/Col';
+
+const Home = () => {
+  const { t } = useTranslation();
+
+  return (
+    <Container>
+      <Row className="justify-content-center">
+        <Col>
+          <img src="https://mysuma.org/wp-content/uploads/2020/06/finalpng-resize.png" alt="MySuma Logo"></img>
```
Comment (rgalanakis) [inline webapp/src/components/Home.js:16]: Download this and include/import it as an asset.

File: webapp/src/index.js [hunk 8]
```diff
@@ -1,17 +1,14 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import App from './components/App';
+import './localization/i18n';
 import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+// Bootstrap Stylesheet
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (rgalanakis) [inline webapp/src/index.js:7]: We need to have our own index we import files through, I'll show you this week.

File: webapp/public/jsonUserList.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+        "phone": "19192224444",
+        "token": "123456"
+    },
+    {
+        "id": 3,
+        "name": "Miley",
+        "phone": "19192225555",
+        "token": "123456"
+    }
+]
```
Comment (DeeTheDev) [inline webapp/public/jsonUserList.json:20]: We can remove this temporary file.

**Pull request merged**